src/node/node_state.h

```diff
@@ -2905,5 +2937,59 @@ namespace ccf
     {
       return writer_factory;
     }
+
+    std::optional<LedgerSecretPtr> try_unseal_ledger_secret()
+    {
+      if (!config.recover.sealed_ledger_secret.has_value())
+      {
+        return std::nullopt;
+      }
+
+      auto sealing_key = ccf::pal::snp::is_sev_snp() ?
+        ccf::pal::snp::make_derived_key()->get_raw() :
+        std::vector<uint8_t>(32, 0);
+      LedgerSecret unsealed_ledger_secret;
+      try
+      {
+        auto raw = crypto::aes_gcm_decrypt(
+          sealing_key, config.recover.sealed_ledger_secret.value());
+        auto json = nlohmann::json::parse(raw);
+        from_json(json, unsealed_ledger_secret);
+      }
+      catch (const std::exception& e)
+      {
+        LOG_INFO_FMT("Failed to unseal full recovery share: {}", e.what());
```

Comment:
```suggestion
        LOG_FAIL_FMT("Failed to unseal full recovery share: {}", e.what());
```

At the least, this should be a `LOG_FAIL`. And as said in the other comment, I think we should avoid returning a nullopt and instead this is probably a "abandon recovery" exception.